### DISCUSSION TITLE
Desktop Shortcut (Enabled/Disabled)

### DISCUSSION THREAD

**Discussion by alevalesantos** [Romanitho/Winget-AutoUpdate#655] [Ideas]
Desktop Shortcut
When the option are setup to  "Enabled", it creates the shortcut on the desktop, but when changing it to "Disabled", it does not remove the desktop shortcut

Suggestion: 
Would be to change it to "disabled", the shortcut should be removed.

**Comment by KnifMelti** [comment]
Settings are applied at installation time.
If **WAU** is **GPO-managed** settings are checked and acted upon **once a day** via the scheduled task `Winget-AutoUpdate-Policies`.
I for one wants the GPO checked every time **WAU** is started, so in `_WAU-mods.ps1` I have:
```
# Use a single call to Get-ScheduledTask and store the result in a variable
$GPOTask = Get-ScheduledTask -TaskName 'Winget-AutoUpdate-Policies' -ErrorAction SilentlyContinue

# Evaluate GPO settings (and take control of the task "Winget-AutoUpdate-Policies")
if ($($WAUConfig.WAU_ActivateGPOManagement -eq 1)) {
    if ($GPOTask) {
        # Check if the task is disabled
        if ($GPOTask.State -eq 'Disabled') {
            # Enable the task and run it
            Enable-ScheduledTask -TaskName 'Winget-AutoUpdate-Policies' -TaskPath 'WAU' | Out-Null
            Write-ToLog "-> Enable task and evaluate GPO settings..." "Yellow"
            $GPOTask | Start-ScheduledTask | Out-Null
        } else {
            # Run the task
            Write-ToLog "-> Evaluate GPO settings..." "Yellow"
            $GPOTask | Start-ScheduledTask | Out-Null
        }

        # Wait until the task finishes running
        while ((Get-ScheduledTask -TaskName 'Winget-AutoUpdate-Policies' -ErrorAction SilentlyContinue).State -eq 'Running') {
            Start-Sleep -Milliseconds 500
        }

        # Disable the task
        Disable-ScheduledTask -TaskName 'Winget-AutoUpdate-Policies' -TaskPath 'WAU' | Out-Null
        Write-ToLog "-> ...disable task!" "Yellow"
    }
}
```
NB. managing via **GPO** creates/deletes the **Desktop Shortcut** just fine.

**Reply by Romanitho** [reply]
In the new MSI installer, shortcut will be managed with the msi itself. Probably, the gpo will be deprecated, and shortcut will have to be managed via msi arguments or manually (or handled by script/mods).
Moreover, in the future, the shortcut could be translated in different languages for example.
